tooling/build/scripts/publishing/index.ts

```diff
@@ -0,0 +1,188 @@
+}
+
+// Wrapper function for debug logging
+function logDebug(message: string, ...optionalParams: any[]) {
+  if (process.env.DEBUG === "true") {
+    console.log(message, ...optionalParams);
+  }
+}
+
+async function main() {
+  const client = new Client({
+    user: DB_USERNAME,
+    host: DB_HOST,
+    database: DB_NAME,
+    password: DB_PASSWORD,
+    port: Number(DB_PORT),
+  });
+
+  const start = performance.now(); // Start profiling
+
+  try {
+    await client.connect();
+
+    // Fetch and write navbar, footer, and config JSONs
+    await fetchAndWriteSiteData(client);
+
+    // Fetch all resources and their full permalinks
+    const resources = await getAllResourcesWithFullPermalinks(client);
+
+    // Process each resource
+    for (const resource of resources) {
+      logDebug(
+        `Processing resource with id ${resource.id}, fullPermalink: ${resource.fullPermalink}`
+      );
+      if (resource.type === "Page" && resource.content) {
+        await writeContentToFile(
+          resource.fullPermalink,
+          resource.content,
+          resource.parentId
+        );
+      } else {
+        logDebug(
+          `Skipping resource with id ${resource.id} as it is not a Page or has no content.`
+        );
+      }
+    }
+  } finally {
+    await client.end();
+    const end = performance.now(); // End profiling
+    console.log(`Program completed in ${(end - start) / 1000} seconds`);
+  }
+}
+
+async function getAllResourcesWithFullPermalinks(
+  client: Client
+): Promise<Resource[]> {
+  const values = [SITE_ID];
+
+  try {
+    const res = await client.query(
+      GET_ALL_RESOURCES_WITH_FULL_PERMALINKS,
+      values
+    );
+    logDebug("Fetched resources with full permalinks:", res.rows);
+    return res.rows;
+  } catch (err) {
+    console.error("Error fetching resources:", err);
+    return [];
+  }
+}
+
+async function writeContentToFile(
+  fullPermalink: string | undefined,
+  content: any,
+  parentId: number | null
+) {
+  try {
+    if (!fullPermalink) {
+      console.error("Error: fullPermalink is undefined or empty for resource");
+      return;
+    }
+
+    let directoryPath: string;
+    let fileName: string;
+
+    if (parentId === null) {
+      // No enclosing folder if parentId is null
+      directoryPath = path.join(__dirname, "schema");
+      fileName = `${fullPermalink}.json`;
+    } else {
+      directoryPath = path.join(
+        __dirname,
+        "schema",
+        path.dirname(fullPermalink)
+      );
+      fileName = `${path.basename(fullPermalink)}.json`;
+    }
+
+    if (!directoryPath || !fileName) {
+      console.error("Error: directoryPath or fileName is undefined");
+      return;
+    }
+
+    const filePath = path.join(directoryPath, fileName);
+
+    // Create directories if they don't exist
+    fs.mkdirSync(directoryPath, { recursive: true });
+
+    // Write JSON content to file
+    fs.writeFileSync(filePath, JSON.stringify(content, null, 2), "utf-8");
```

Comment:
nit: I think we can combine this with `writeJsonToFile`?

Reply:
the logic is a bit different for writing to schema vs data folders, preferring to refactor this later if needed